Moderation Matrix (#87)
TheOdinProject/top-meta: Moderation_Matrix.md

# Moderation Matrix

## 1 Zap (WARNING) 
Minor violations such as:
- "Mini-modding"
  - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
- Asking about topics or projects unrelated to The Odin Project (Rules 11 and 12)
-Mild unprofessionalism: see this guide https://www.bitrix24.com/about/blogs/communications/instant-messaging-and-group-chat-etiquette-in-a-professional-environme.php 
  - Drug related conversation (Rule 7)

>   - Special attention to treating everyone with respect

Reviewer: ```suggestion
  - Treating others on the server disrespectfully
```
Shouldn't this be invested since we're outlining things that are violations?

Author: Agree

  - Mental health issues discussion (Rule 12)
- A second infraction of the following:
  - Short circuiting a teaching moment (Rule 10)
- Asking for help with Windows or encouraging the use of it
  - “Chat bombing”, aka posting many short (ususally 4-5 or fewer words) messages in rapid succession, repeatedly sending parts of thoughts (incomplete, non-standalone sentences or fragments) in new messages. (Use shift + enter and send ONE message where possible or edit your message) (Rule 7)

## 2 Zaps (WARNING)
Medium severity violations such as: 
- Suggesting piracy (Rule 9)
- Unsolicited pings/DM/friend request (Rule 6)
- Political or religious topics (Rule 8)
- Mild toxicity (directed insults, provoking/baiting other users, passive aggression, naming & shaming) (Rule 7)
- Self promo without permission (ask with Modmail)(Rule 7)

>   - Do not share resources you have created solely for personal or monetary gain

Reviewer 2: Can I get clarity on what is intended by resources created solely for personal gain?

Author: I wasn't a part of that part of the convo, but I took it as people who are blogging to build a platform, and wanting to share stuff here for a larger audience and there's zero relation to TOP or issues a learner might be encountering.

Reviewer 3: @twalton83 @CouchofTomato 

The discussion was mostly about people that show up in the server simply to plug their stuff. It could be their programming server, they're own content, or anything.

It's to prevent people who just shamelessly self-promote from permeating our server. It's a form of spam and free advertisement.

Reviewer 2: Right. That makes sense. Thanks.

  - Examples: "Fill out my survey" (does not help the goals of Odin), "Join my boot camp/program/server" (Could mislead learners), "Why PHY is superior" (unrelated blog post)

## 5 Zaps
High severity violations such as: 
- Posting about any illegal activities, such as facilitating piracy (Rule 7 and 9)
- Publicly arguing about receiving moderation activity after being asked to take it to ModMail/discuss privately 
- Excessive toxicity (targeted harassment, inciting drama, aggressively arguing purposefully, etc.) (Rule 7)

## 10 Zaps (Immediate Ban)
Extreme severity violations such as: 
- Bigotry (racism, homophobia, hate speech etc) (Rule 7 and 8)
- Harassment (Rule 7)
- NSFW Imagery (Rule 7)
- Spamming 
  - Discord Nitro spam
  - Self promotion (joining just to promote material, no prior chat history)
  - Repeated random nonsensical messages (copypasta)
- Doxxing (Rule 7)

> 

Reviewer: I feel like we will probably want to throw a catch-all under here for when people do something not explicitly outlined but is deserving of a ban. Along the lines of "Any other behavior deemed unacceptable and deserving of a ban by the moderation team".

> Zaps expire at a rate of 1 per week. 

Reviewer: Do multi-zap offenses expire all at once (i.e. a 5 zap expiring all zaps after 5 weeks) or does it tick down one per week regardless of how the zaps were accrued?

Reviewer 3: I don't know if this was discussed, the rules we based it from did state they would drop off once per week. However, I'd maybe say "an infraction per week" would be removed if we expected 5 to be removed at once.

Author: I intended this as a higher violation takes longer to fall off because it's more severe. 5 zap infraction goes to 4, goes to 3, etc. Every infraction falling off at the same rate (just a week) doesn't make sense to me personally. It's basically giving the same punishment.

# Moderation Matrix

> 

Reviewer 4: A link to the Discord rules here could be useful as a reference (do they exist on a repo somewhere? Couldn't find them - I can add if needed)

Reviewer 5: I think we should be using the newly updated rules that are available on our website. https://www.theodinproject.com/community_rules. Matter of fact, due to a recent conversation that I was having with maintainers, I think the final home of this matrix and these community rules needs to be one the same page. (This will reduce the places to update rules in the future and make it less information for users to read).

## 1 Zap (WARNING) 
Minor violations such as:
- "Mini-modding"
  - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
- Asking about topics or projects unrelated to The Odin Project (Rules 11 and 12)

> -Mild unprofessionalism: see this guide https://www.bitrix24.com/about/blogs/communications/instant-messaging-and-group-chat-etiquette-in-a-professional-environme.php 

Reviewer 4: ```suggestion
- Mild unprofessionalism: see this [guide](https://www.bitrix24.com/about/blogs/communications/instant-messaging-and-group-chat-etiquette-in-a-professional-environme.php)
```
Minor typo breaking list + improve link formatting

Reviewer 6: In relation to @rlmoser99 comment below about sending users to this external link, if it *is* decided to keep this link, the verbiage/link text I would say should instead be, `see this guide on [Group Chat Etiquette in a Professional Environment]`

> -Mild unprofessionalism: see this guide https://www.bitrix24.com/about/blogs/communications/instant-messaging-and-group-chat-etiquette-in-a-professional-environme.php 

Reviewer 5: In thinking about this, I am not a fan of sending users to another link, especially because several of these points are covered in our new rules under 10 (Be professional and have good etiquette). Instead, I think we should take the few things from this article not already covered and add it to our list.

Author: My thought is we already send people to external link in our rules to begin with.

Reviewer 3: If we have our own list for it, it may be worth putting that link in our rules if is not fully covered.

  - Drug related conversation (Rule 7)
  - Special attention to treating everyone with respect

>   - Mental health issues discussion (Rule 12)

Reviewer 5: I am not sure that this fits under "Mild unprofessionalism" and should also mention relationships and depression. In our rules, we have it under the "This community only supports the content in our curriculum" rule. However, if we feel strongly that it needs to stay here, then I wonder if we should also move it in our rules.

I also think it should be moved under the "A second infraction" or 0 zaps for first-time offenders.

Author: I think depression falls under the last point. For me, all of these topics would probably cause discomfort in the workplace.

Reviewer 3: I agree @twalton83 , I had a talk about this with my manager at work and she did say that people airing our their personal issues in a semi-public place where people have to do business is unprofessional. Relationships should for sure be added.

> - A second infraction of the following:

Reviewer 4: Since 1 zap ouchies are minor by definition, we should consider applying the 0 zap second infraction rule to all of them instead of just these ones. 

This is simpler, more consistent and slightly friendlier to otherwise well meaning and / or anxious newbs.

If every innocuous offense receives a zap immediately there's a risk that it discourages engagement / participation by new people, many of whom are already hesitant / apologetic when asking their first questions

Reviewer 5: I'm not sure if they should all be 0 zaps, but I do like the idea of having these be a list of 0 zaps, because I think the wording "A second infraction of the following" can be somewhat confusing to people. Especially since how a second infraction of all of them will be given an increase of zaps for a second infraction. 

I think we have to zap people who mini mod, Drug related conversation, and not treating people with respect.

Reviewer: I also would support a 0 zap category for these items.

  - Short circuiting a teaching moment (Rule 10)

>   - Windows (Rule 14)
>     - Asking for help on or with
>     - Encouraging use of

Reviewer 5: I think instead of sub-bullet points here, we should put it together in one cohesive point.
```suggestion
- Asking for help with Windows or encouraging the use of it
```

> - “Chat bombing”, aka posting many short (ususally 4-5 or fewer words) messages in rapid succession, repeatedly sending parts of thoughts (incomplete, non-standalone sentences or fragments) in new messages. (Use shift + enter and send ONE message where possible or edit your message) (Rule 7)

Reviewer 5: From the discussions the other day, I think this should be below the "A second infraction of the following:" (or be moved to a 0 point area).
```suggestion
  - “Chat bombing”, aka posting many short (ususally 4-5 or fewer words) messages in rapid succession, repeatedly sending parts of thoughts (incomplete, non-standalone sentences or fragments) in new messages. (Use shift + enter and send ONE message where possible or edit your message) (Rule 7)
```

## 2 Zaps (WARNING)
Medium severity violations such as: 
- Suggesting piracy (Rule 9)
- Unsolicited pings/DM/friend request (Rule 6)
- Political or religious topics (Rule 8)
- Mild toxicity (directed insults, provoking/baiting other users, passive aggression, naming & shaming) (Rule 7)
- Self promo without permission (ask with Modmail)(Rule 7)
  - Do not share resources you have created solely for personal or monetary gain
  - Examples: "Fill out my survey" (does not help the goals of Odin), "Join my boot camp/program/server" (Could mislead learners), "Why PHY is superior" (unrelated blog post)

## 5 Zaps
High severity violations such as: 
- Posting about any illegal activities, such as facilitating piracy (Rule 7 and 9)
- Publicly arguing about receiving moderation activity after being asked to take it to ModMail/discuss privately 
- Excessive toxicity (targeted harassment, inciting drama, aggressively arguing purposefully, etc.) (Rule 7)

## 10 Zaps (Immediate Ban)
Extreme severity violations such as: 

> - Bigotry (racism, homophobia, etc) (Rule 7 and 8)

Reviewer 4: ```suggestion
- Bigotry (racism, homophobia, hate speech etc) (Rule 7 and 8)
```
Obviously a minor distinction but doesn't hurt. Covers things like Hitler avatars / references.

- Harassment (Rule 7)
- NSFW Imagery (Rule 7)
- Spamming 
  - Discord Nitro spam
  - Self promotion (joining just to promote material, no prior chat history)
  - Repeated random nonsensical messages (copypasta)
- Doxxing (Rule 7)

Zaps expire at a rate of 1 per week. 

Repeated offenses with unexpired "zaps" increase to the next penalty tier. E.g. The second time a user is warned about unsolicited pings or DMs, they earn 5 zaps instead of 2, making them 3 zaps away from a ban.

This is not a comprehensive list and zaps are at the discretion of the moderation team. Please DM @ModMail if you would like to discuss this matrix or any moderation activity (whether you are on the receiving end or not). 

> 

Reviewer 4: Nice work 👍

# Moderation Matrix
This matrix correlates with the [Community Rules](https://www.theodinproject.com/community_rules) that are required to be followed in order to participate in the community. Some rules have different levels of severity, so you may see references to them under different numbers of zaps. 

This matrix is not all encompassing and zaps are given at the discretion of the moderator team.

## 0 Zaps (Heads Up)
- Dog piling/short circuiting a teaching moment
- Asking about or encouraging the use of Windows
- "Chat bombing", aka posting many short (ususally 4-5 or fewer words) messages in rapid succession, repeatedly sending parts of thoughts (incomplete, non-standalone sentences or fragments) in new messages. (Use shift + enter and send ONE message where possible or edit your message) 

> - Discussing mental health issues in depth

Reviewer 6: Might be covered enough by the "Asking about topics unrelated to TOP" one, but maybe including "or seeking mental health help from other users"?

## 1 Zap (WARNING) 
Minor violations such as:
- "Mini-modding"
  - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
- Asking about topics or projects unrelated to The Odin Project 
-Mild unprofessionalism:
  - Drug related conversation 
  - Treating others on the server disrespectfully

>   - Mental health issues discussion 

Reviewer 6: I'd say this should be removed from the 1 zap category since it's already covered by 0 zaps. Then just tweak the 0 zap verbiage to "Discussing mental health issues in depth or seeking help with mental health from other users" or something.

  - Relationship advice
- A second infraction of any 0 Zap issues above

## 2 Zaps (WARNING)
Medium severity violations such as: 
- Suggesting piracy 
- Unsolicited pings/DM/friend request 
- Political or religious topics 
- Mild toxicity (directed insults, provoking/baiting other users, passive aggression, naming & shaming)
- Self promo without permission (ask with Modmail)
  - Do not share resources you have created solely for personal or monetary gain
  - Examples: "Fill out my survey" (does not help the goals of Odin), "Join my boot camp/program/server" (Could mislead learners), "Why PHP is superior" (unrelated blog post)

## 5 Zaps
High severity violations such as: 
- Posting about any illegal activities, such as facilitating piracy 
- Publicly arguing about receiving moderation activity after being asked to take it to ModMail/discuss privately 
- Excessive toxicity (targeted harassment, inciting drama, aggressively arguing purposefully, etc.) 

## 10 Zaps (Immediate Ban)
Extreme severity violations such as: 
- Bigotry (racism, homophobia, hate speech etc)
- Harassment 

> - NSFW Imagery

Reviewer 6: Thinking this might be better as "NSFW or otherwise highly offensive imagery in posts, profile pictures, or usernames", or something along those lines?

# Moderation Matrix
This matrix correlates with the [Community Rules](https://www.theodinproject.com/community_rules) that are required to be followed in order to participate in the community. Some rules have different levels of severity, so you may see references to them under different numbers of zaps. 

This matrix is not all encompassing and zaps are given at the discretion of the moderator team.

## 0 Zaps (Heads Up)
- Dog piling/short circuiting a teaching moment
- Asking about or encouraging the use of Windows
- "Chat bombing", aka posting many short (ususally 4-5 or fewer words) messages in rapid succession, repeatedly sending parts of thoughts (incomplete, non-standalone sentences or fragments) in new messages. (Use shift + enter and send ONE message where possible or edit your message) 
- Discussing mental health issues in depth

> 

Reviewer 7: ```suggestion

```

## 1 Zap (WARNING) 
Minor violations such as:
- "Mini-modding"

>   - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community

Reviewer 6: Minor: I think two separate bullet points for "leave moderation" and "use ModMail" might work better

> - Asking about topics or projects unrelated to The Odin Project 

Reviewer 6: Minor: I'm thinking this could be improved somehow. Maybe, "Asking for advice on projects unrelated to The Odin Project, or asking about topics unrelated to our Discord channels"?

Main thinking is I can see people questioning the point of games/movies/off topic channels if they're unrelated to TOP, but this could also be thinking this would be a bigger deal than it actually would be.

> -Mild unprofessionalism:

Reviewer 7: ```suggestion
- Mild unprofessionalism:
```

> - "Mini-modding"
>   - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
> - Asking about topics or projects unrelated to The Odin Project 
> -Mild unprofessionalism:
>   - Drug related conversation 
>   - Treating others on the server disrespectfully
>   - Mental health issues discussion 
>   - Relationship advice
> - A second infraction of any 0 Zap issues above

Reviewer 7: ```suggestion
- A second infraction of any 0 Zap issues above
- "Mini-modding"
  - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
- Asking about topics or projects unrelated to The Odin Project 
-Mild unprofessionalism:
  - Drug related conversation 
  - Treating others on the server disrespectfully
  - Mental health issues discussion 
  - Relationship advice
```

> Minor violations such as:
> - "Mini-modding"
>   - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
> - Asking about topics or projects unrelated to The Odin Project 
> -Mild unprofessionalism:
>   - Drug related conversation 
>   - Treating others on the server disrespectfully
>   - Mental health issues discussion 
>   - Relationship advice
> - A second infraction of any 0 Zap issues above

Reviewer 7: This looks super confusing in the preview, but it just moves the "second infraction of 0 zap" thing to the top. I think it reads better, especially because those are still fresh in the reader's mind.
```suggestion
Minor violations such as:
- A second infraction of any 0 Zap issues above
- "Mini-modding"
  - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
- Asking about topics or projects unrelated to The Odin Project 
- Mild unprofessionalism:
  - Drug related conversation 
  - Treating others on the server disrespectfully
  - Mental health issues discussion 
  - Relationship advice
```

## 2 Zaps (WARNING)
Medium severity violations such as: 

> - Suggesting piracy 

Reviewer 6: Should "suggesting piracy" really be separate from "facilitating it"? Not a hill I'll die on, but am curious if it wouldn't be better to have any form of piracy at a single level rather than split out like this.

Reviewer 6: Definitely true. Mainly wanted to see if we wanted to keep them as separate offenses or just lump them together as a general "no piracy" rule. But again, not a hill I'll die on and I'm good with things as-is.

- Unsolicited pings/DM/friend request 
- Political or religious topics 

> - Mild toxicity (directed insults, provoking/baiting other users, passive aggression, naming & shaming)

Reviewer 7: ```suggestion
- Mild toxicity
  - Directed insults
  - Provoking/baiting other users
  - Passive aggression
  - Naming & shaming
```

- Self promo without permission (ask with Modmail)
  - Do not share resources you have created solely for personal or monetary gain
  - Examples: "Fill out my survey" (does not help the goals of Odin), "Join my boot camp/program/server" (Could mislead learners), "Why PHP is superior" (unrelated blog post)

> ## 5 Zaps

Reviewer 7: ```suggestion
## 5 Zaps (SEVERE WARNING)
```

High severity violations such as: 
- Posting about any illegal activities, such as facilitating piracy 
- Publicly arguing about receiving moderation activity after being asked to take it to ModMail/discuss privately 
- Excessive toxicity (targeted harassment, inciting drama, aggressively arguing purposefully, etc.) 

## 10 Zaps (Immediate Ban)
Extreme severity violations such as: 
- Bigotry (racism, homophobia, hate speech etc)

> - Harassment 

Reviewer 6: Considering 5 zaps already has "targetted harassment", this might be worth including examples of how this would differ from that.

Reviewer 8: Good point ☝️

- NSFW Imagery
- Spamming 
  - Discord Nitro spam
  - Self promotion (joining just to promote material, no prior chat history)
  - Repeated random nonsensical messages (copypasta)
- Doxxing 
- Any other behavior deemed unacceptable and deserving of a ban by the moderation team

Zaps decay at a rate of 1 zap per week. E.g., if you receive 5 zaps on a Friday, the following Friday you will have 4, the Friday after that you will have 3, and so forth. This means higher level infractions may impact your ability to participate for a longer period of time if further rules are broken. 

Repeated offenses with unexpired "zaps" increase to the next penalty tier. E.g. The second time a user is warned about unsolicited pings or DMs, they earn 5 zaps instead of 2, making them 3 zaps away from a ban.

This is not a comprehensive list and zaps are at the discretion of the moderation team. Please DM @ModMail if you would like to discuss this matrix or any moderation activity (whether you are on the receiving end or not). 

> 

Reviewer 7: ```suggestion

```

- Discussing mental health issues in depth

## 1 Zap (WARNING) 
Minor violations such as:
- A second infraction of any 0 Zap issues above
- "Mini-modding"
  - Please leave moderation action to the moderators, use ModMail to report rule breaking or if you see something you’re unsure about in the community
- Asking about topics or projects unrelated to The Odin Project 
- Mild unprofessionalism:
  - Drug related conversation 
  - Treating others on the server disrespectfully
  - Mental health issues discussion 
  - Relationship advice

## 2 Zaps (WARNING)
Medium severity violations such as: 
- Suggesting piracy 
- Unsolicited pings/DM/friend request 
- Political or religious topics 
- Mild toxicity:
  - Directed insults
  - Provoking/baiting other users
  - Passive aggression
  - Naming & shaming
- Self promo without permission (ask with Modmail)
  - Do not share resources you have created solely for personal or monetary gain
  - Examples: "Fill out my survey" (does not help the goals of Odin), "Join my boot camp/program/server" (Could mislead learners), "Why PHP is superior" (unrelated blog post)

> ## 5 Zaps (SEVERE WARNING)

Reviewer: Did we decide against doing temp bans and/or timeouts at certain zap levels pre-permaban?

Author: Nope, implementation itself has not been determined.

Reviewer 6: I could be in favor of this level of zaps resulting in a temp ban (length TBD). At this point, a user would have accumulated enough zaps and thus had enough warnings regarding behavior, that a temp ban would give them time to reflect without causing any further issues on the server.

I will say I can also see the drawback to this. If someone has had zero issues before and out of nowhere links to a pirated source, should this first ever offense of a user result in a temp ban?

Author: Yeah that's my thing, it would be difficult to be consistent here.

Reviewer: I don't think this is something that we need to hammer out before we can move forward by any means. I also wouldn't be against temp bans just being a matter of moderator discretion.

- Asking for advice/help on projects unrelated to The Odin Project, or asking about topics unrelated to our Discord channels
- Mild unprofessionalism:
  - Drug related conversation 
  - Treating others on the server disrespectfully
  - Relationship advice

## 2 Zaps (WARNING)
Medium severity violations such as: 
- Suggesting piracy 
- Unsolicited pings/DM/friend request 
- Political or religious topics 
- Mild toxicity:
  - Directed insults
  - Provoking/baiting other users
  - Passive aggression
  - Naming & shaming
- Self promo without permission (ask with Modmail)
  - Do not share resources you have created solely for personal or monetary gain
  - Examples: "Fill out my survey" (does not help the goals of Odin), "Join my boot camp/program/server" (Could mislead learners), "Why PHP is superior" (unrelated blog post)

## 5 Zaps (SEVERE WARNING)
High severity violations such as: 
- Posting about any illegal activities, such as facilitating piracy 
- Publicly arguing about receiving moderation activity after being asked to take it to ModMail/discuss privately 
- Excessive toxicity (targeted harassment, inciting drama, aggressively arguing purposefully, etc.) 

> ## 10 Zaps (Immediate Ban)

Reviewer 6: Something I hadn't though of in my previous review, but would this level of zaps result in a *permanent* ban, or would the length of the ban depend on the situation?

This may not matter *too* much if we don't plan on having any other level of zaps result in a temp ban or anything (going off a previous comment from Sully that I will also comment on), but thought it might be worth thinking about/a discussion.

Author: It's permanent imo- these are all things that would bar you from participating ever again, and if you're continually breaking rules and rack up this many points, you aren't being a good member of this community.
